components/user-store/org.wso2.carbon.identity.user.store.count/src/main/java/org/wso2/carbon/identity/user/store/count/UserStoreCountService.java

```diff
@@ -41,26 +48,22 @@
      * @return the number of users matching the filter by each domain
      */
     public PairDTO[] countUsers(String filter) throws UserStoreCounterException {
+
         Set<String> userStoreDomains = UserStoreCountUtils.getCountEnabledUserStores();
         PairDTO[] userCounts = new PairDTO[userStoreDomains.size()];
         int i = 0;
 
         for (String userStoreDomain : userStoreDomains) {
-            UserStoreCountRetriever counter = UserStoreCountUtils.getCounterInstanceForDomain(userStoreDomain);
-            Long count = Long.valueOf(-1);
-            if (counter != null) {
-                try {
-                    count = counter.countUsers(filter);
-                } catch (UserStoreCounterException e) {
-                    log.error("Error while getting user count from user store domain : " + userStoreDomain, e);
-                }
-            } else {
-                //no action
+            String filterWithDomain = getFilterWithDomain(userStoreDomain, filter);
+            long count = -1L;
+            try {
+                count = getUserCountWithClaims(UserStoreCountUtils.USERNAME_CLAIM, filterWithDomain);
+            } catch (UserStoreCounterException e) {
+                log.error("Error while getting user count from user store domain : " + userStoreDomain, e);
```

Comment:
The exception is ignored here(only an error log is present). Shouldn't we pass the exception without handling it? Any reason just to have a log?